base_multi_image/models/image.py

```diff
@@ -0,0 +1,181 @@
+    _name = "base_multi_image.image"
+    _sql_constraints = [
+        ('uniq_name_owner', 'UNIQUE(owner_id, owner_model, name)',
+         _('A document can have only one image with the same name.')),
+    ]
+
+    owner_id = fields.Integer(
+        "Owner",
+        required=True)
+    owner_model = fields.Char(
+        required=True)
+    storage = fields.Selection(
+        [('url', 'URL'), ('file', 'OS file'), ('db', 'Database')],
+        required=True,
+        default='db')
+    name = fields.Char(
+        'Image title',
+        required=True,
+        translate=True)
+    filename = fields.Char()
+    extension = fields.Char(
+        'File extension',
+        readonly=True)
+    file_db_store = fields.Binary(
+        'Image stored in database',
+        filters='*.png,*.jpg,*.gif')
+    path = fields.Char(
+        "Image path",
+        help="Image path")
+    url = fields.Char(
+        'Image remote URL')
+    image_main = fields.Binary(
+        "Full-sized image",
+        compute="_get_image")
+    image_medium = fields.Binary(
+        "Medium-sized image",
+        compute="_get_image_sizes",
+        help="Medium-sized image. It is automatically resized as a "
+             "128 x 128 px image, with aspect ratio preserved, only when the "
+             "image exceeds one of those sizes. Use this field in form views "
+             "or kanban views.")
+    image_small = fields.Binary(
+        "Small-sized image",
+        compute="_get_image_sizes",
+        help="Small-sized image. It is automatically resized as a 64 x 64 px "
+             "image, with aspect ratio preserved. Use this field anywhere a "
+             "small image is required.")
+    comments = fields.Text(
+        'Comments',
+        translate=True)
+    show_technical = fields.Boolean(
+        compute="_show_technical")
+
+    @api.multi
+    @api.depends('storage', 'path', 'file_db_store', 'url')
+    def _get_image(self):
+        """Get image data from the right storage type."""
+        for s in self:
+            s.image_main = getattr(s, "_get_image_from_%s" % s.storage)()
+
+    @api.multi
+    @api.depends("owner_id", "owner_model")
+    def _show_technical(self):
+        """Know if you need to show the technical fields."""
+        self.show_technical = all(
+            "default_owner_%s" % f not in self.env.context
+            for f in ("id", "model"))
+
+    @api.multi
+    def _get_image_from_db(self):
+        return self.file_db_store
+
+    @api.multi
+    def _get_image_from_file(self):
+        if self.path and os.path.exists(self.path):
+            try:
+                with open(self.path, 'rb') as f:
+                    return base64.b64encode(f.read())
+            except Exception as e:
+                _logger.error("Can not open the image %s, error : %s",
+                              self.path, e, exc_info=True)
+        else:
+            _logger.error("The image %s doesn't exist ", self.path)
+
```

Comment:
> Nice catch. However, when you use a file image here, it's expected that the image is already in the system. You just provide the path and Odoo reads it. No file saving on disk.

> in this case Odoo doesn't handle that. This mode is thought for sharing the same files for the e-commerce/mediastore and Odoo, so you don't have to duplicate the storage.

I get it! So that's fine :)